index/dlg_index.c

```diff
@@ -2429,6 +2429,33 @@ struct Mailbox *mutt_index_menu(struct MuttWindow *dlg, struct Mailbox *m_init)
         change_folder_notmuch(priv->menu, buf, sizeof(buf), &priv->oldcount, shared, false);
         break;
       }
+
+      case OP_MAIN_WINDOWED_VFOLDER_RESET:
+      {
+        if (!prereq(shared->ctx, priv->menu, CHECK_IN_MAILBOX))
+          break;
+        const short c_nm_query_window_duration =
+            cs_subset_number(shared->sub, "nm_query_window_duration");
+        const bool c_nm_query_window_enable =
+            cs_subset_bool(shared->sub, "nm_query_window_enable");
+        if (!c_nm_query_window_enable && (c_nm_query_window_duration <= 0))
```

Comment:
> nm_query_window_duration is marked DT_NOT_NEGATIVE, so the <=0 could be ==0

That conditional pre-dates the introduction configuration validation framework. It used to be possible to set a negative value without NeoMutt's complaint. It's better to leave the working code alone than make modifications as the initial implementation assumptions may differ from our current understanding.

> Should that be || instead of &&

No, that would disable query windows whenever `nm_query_window_enable=false` or `nm_query_window_duration=0`. 

This is not desired behavior as `nm_query_window_enable` is a new config. variable to allow query windows when `nm_query_window_duration=0`. This allows windows over a single timebase (e.g. `date:1month..1month`). Before that the explicit enable `nm_query_window_duration=0` disabled the query windows.

It was implemented this way so we wouldn't break existing user setups while supporting new functionality. It's a better user experience.

Reply:
okay, I missed DeMorgan's rules at a distance. The new code is !(call()), where call does 'enable || duration &gt; 0)', but moving the ! inside the call is indeed (!enable) && (!(duration &gt; 0)).